crates/turborepo-repository/src/change_mapper/mod.rs

```diff
@@ -61,31 +70,40 @@ impl<'a, PD: PackageChangeMapper> ChangeMapper<'a, PD> {
         lockfile_change: Option<LockfileChange>,
     ) -> Result<PackageChanges, ChangeMapError> {
         if Self::default_global_file_changed(&changed_files) {
-            return Ok(PackageChanges::All);
+            return Ok(PackageChanges::All(
+                AllPackageChangeReason::DefaultGlobalFileChanged,
+            ));
         }
 
         // get filtered files and add the packages that contain them
         let filtered_changed_files = self.filter_ignored_files(changed_files.iter())?;
-        let PackageChanges::Some(mut changed_pkgs) =
-            self.get_changed_packages(filtered_changed_files.into_iter())
-        else {
-            return Ok(PackageChanges::All);
-        };
```

Comment:
I wanted to pass through the reason to `PackageChanges::All` here, so converted this to a longer `match` resulting in a larger diff in this file.